src/decoder/decoder-wrappers.cc

```diff
@@ -398,7 +398,7 @@ template bool DecodeUtteranceLatticeIncremental(
     double *like_ptr);
 
 template bool DecodeUtteranceLatticeIncremental(
-    LatticeIncrementalDecoderTpl<fst::GrammarFst> &decoder,
+    LatticeIncrementalDecoderTpl<fst::GrammarFstTpl<const fst::ConstFst<fst::StdArc> > > &decoder,
```

Comment:
A declaration like `using GrammarFst = GrammarFstTpl<const fst::ConstFst<fst::StdArc> >` will reduce the number of changes like this.

Reply:
Agree on the `using`.The reason for wanting to instantiate it with the concrete types of that OpenFST has some magic so that if the concrete type is known at compile time, it can avoid virtual function lookups..